src/main/java/com/example/planit/engine/AdminEngine.java

```diff
@@ -146,4 +151,67 @@ public DTOcoursesResponseToController getCourseFromDB(String courseId, String su
             return new DTOcoursesResponseToController(false, "Error fetching course from database: " + e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
         }
     }
+
+    public DTOusersResponseToController getAllUsersFromDB(String sub) {
+        try {
+            Optional<User> maybeUser = userRepo.findUserBySubjectID(sub);
+
+            if (maybeUser.isEmpty()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_USER_NOT_FOUND, HttpStatus.BAD_REQUEST);
+            }
+            User maybeAdminUser = maybeUser.get();
+
+            if (!maybeAdminUser.isAdmin()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_UNAUTHORIZED_USER, HttpStatus.UNAUTHORIZED);
+            }
+
+            List<User> users = userRepo.findAll();
+
+            List<UserClientRepresentation> userClientRepresentations = new ArrayList<>();
+
+            for (User user : users) {
+                userClientRepresentations.add(buildUserClientRepresentationFromUser(user));
+            }
+
+            return new DTOusersResponseToController(true, Constants.NO_PROBLEM, HttpStatus.OK, userClientRepresentations);
+
+        } catch (Exception e) {
+            return new DTOusersResponseToController(false, "Error fetching users from database: " + e.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    public DTOusersResponseToController makeUserAdminInDB(String sub, String userSubId) {
+
+        try {
+
+            Optional<User> maybeUser = userRepo.findUserBySubjectID(sub);
+
+            if (maybeUser.isEmpty()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_USER_NOT_FOUND, HttpStatus.BAD_REQUEST);
+            }
+
+            User maybeAdminUser = maybeUser.get();
+
+            if (!maybeAdminUser.isAdmin()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_UNAUTHORIZED_USER, HttpStatus.UNAUTHORIZED);
+            }
+
+            // Check if user exists in database
+            Optional<User> maybeExistingUser = userRepo.findUserBySubjectID(userSubId);
+            if (maybeExistingUser.isEmpty()) {
+                return new DTOusersResponseToController(false, Constants.ERROR_USER_NOT_FOUND, HttpStatus.BAD_REQUEST);
```

Comment:
The constant : `Constants.ERROR_USER_NOT_FOUND` is returned in the response both when the admin user is not in the DB (line 196)  and when the user to be converted is not in the DB (line 202).

Reply:
The issue #18 will fix that problem, which is in a greater scale, in the future. So I think this PR can be merged.